lib/decode.js

```diff
@@ -4,18 +4,30 @@ var Dict = require("./dict")
  * Decodes bencoded data.
  *
  * @param  {Buffer} data
- * @param  {String} encoding
+ * @param  {Number} start (optional)
+ * @param  {Number} end (optional)
+ * @param  {String} encoding (optional)
  * @return {Object|Array|Buffer|String|Number}
  */
-function decode( data, encoding ) {
-
+function decode( data, start, end, encoding ) {
+  
+  if( typeof start !== 'number' && encoding == null ) {
+    encoding = start
+    start = void 0
```

Comment:
for the sake of readability, don't you think we can just expect 'undefined' to be ... well, undefined?